contracts/adapters/GmpTransferAdapter.sol

```diff
@@ -0,0 +1,147 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import "../../contracts/interfaces/IBridge.sol";
+import "../../contracts/interfaces/IFeeHandler.sol";
+import "../XERC20/interfaces/IXERC20.sol";
+import "./interfaces/IGmpTransferAdapter.sol";
+
+/**
+        .__   __.   ______   .___________. __    ______  _______
+        |  \ |  |  /  __  \  |           ||  |  /      ||   ____|
+        |   \|  | |  |  |  | `---|  |----`|  | |  ,----'|  |__
+        |  . `  | |  |  |  |     |  |     |  | |  |     |   __|
+        |  |\   | |  `--'  |     |  |     |  | |  `----.|  |____
+        |__| \__|  \______/      |__|     |__|  \______||_______|
+
+    Be careful when interacting with this contact as it enables
+    permissionless token addition and transfers via Sygma brige.
+    Always double check contract addresses and code you are interacting with
+    since a malicious actor could deploy a fake contract on a route
+    that is isn't set up by the Sygma team or a trusted 3rd party.
+    This can result in loss of all your funds.
```

Comment:
```suggestion
    Be careful when interacting with this contact as it enables
    permissionless token addition and transfers via Sygma bridge.
    Make sure that a malicious actor cannot deploy arbitrary code
    to the same address as your XERC20 token on another chain.
    This would allow them to burn fake tokens on chain A then mint
    legit tokens on chain B.
```